src/pip/_internal/vcs/git.py

```diff
@@ -48,6 +48,12 @@ class Git(VersionControl):
     def get_base_rev_args(rev):
         return [rev]
 
+    def is_immutable_rev(self, url):
```

Comment:
So based on our discussion on the original issue, I think you want to reframe this not as "is immutable ref" but rather "looks like immutable ref." And the return value would tell you if it's a candidate for checking to see if it's already in the cache and not whether it's something that necessarily can be cached.